src/rust/en.hentaifox/src/lib.rs

```diff
@@ -193,6 +156,27 @@ fn get_page_list(_manga_id: String, chapter_id: String) -> Result<Vec<Page>> {
 		.header("User-Agent", USER_AGENT)
 		.html()?;
 
+	let content = html.select("body > script").first().data().read();
+
+	let start_pattern = r#"parseJSON('"#;
+	let end_pattern = r#"');"#;
+
+	let json = if let Some(start_index) = content.find(start_pattern) {
+		let start_index = start_index + start_pattern.len();
+		if let Some(end_index) = content[start_index..].find(end_pattern) {
+			let end_index = start_index + end_index;
+			Some(&content[start_index..end_index])
+		} else {
+			None
+		}
+	} else {
+		None
+	};
+
+	let json = json
+		.and_then(|json| aidoku::std::json::parse(json).ok())
+		.and_then(|json| json.as_object().ok());
```

Comment:
could probably put this in the first json variable so you don't need to reassign